packages/hydrogen/src/createHydrogenContext.ts

```diff
@@ -239,9 +252,19 @@ export function createHydrogenContext<
     customerAccount,
   });
 
-  // Create React Router context provider
+  // Create React Router context provider (must be synchronous)
   const routerProvider = new RouterContextProvider();
 
+  // Fetch analytics tokens if enabled (async operation)
+  let analyticsTokens: AnalyticsTokens | undefined;
+  if (analyticsOptions?.enabled) {
+    analyticsTokens = await getBackendApprovedTokens(
```

Comment:
Awaiting this here means we are adding a waterfall request (when we need to fetch cookies, which is only the first rendering for a session / new user).
Instead, (as far as I understand without much context) we could run this asynchronously and await for it only when we use the tokens to create the cookie (around [here](https://github.com/Shopify/hydrogen/pull/3144/files#diff-d12882c52d97fd7d9720d5b6d6d2490a733bad02c30d58b553b9631a33982b7aR57), tbd exact location).

This means we run the token request in parallel to other SFR requests and SSR, so when we await for it it should be a no-op. That also simplifies the code a bit because `createHydrogenContext` can keep being synchronous, so we don't need to update many parts of the code adding `await`.